tests/Unit/IO/Observers/Test_RegisterSingleton.cpp

```diff
@@ -0,0 +1,118 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <string>
+#include <type_traits>
+#include <vector>
```

Comment:
`<utility>` for `std::pair`, couldn't find where `<string>`, `<vector>`, or `<type_traits>` were used in this test.